dds/DCPS/DomainParticipantFactoryImpl.cpp

```diff
@@ -174,6 +174,14 @@ DomainParticipantFactoryImpl::delete_participant(
   DomainParticipantImpl* the_servant
   = dynamic_cast<DomainParticipantImpl*>(a_participant);
 
+  if (!the_servant) {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: ")
+      ACE_TEXT("DomainParticipantFactoryImpl::delete_participant: ")
+      ACE_TEXT("failed to obtain the DomainParticipantImpl.\n")));
+
+    return DDS::RETCODE_PRECONDITION_NOT_MET;
```

Comment:
Let's use RETCODE_ERROR here.  The PRECONDITION_NOT_MET error is used for specific preconditions listed in the spec.  If this cast fails there is an internal error in the implementation.